docs/csharp/language-reference/operators/snippets/SwitchExpressions.cs

```diff
@@ -0,0 +1,137 @@
+            Console.WriteLine($"Cardinal orientation is {orientation}");
+            // </SnippetBasicStructure>
+        }
+
+        private static void TypeExample<T>(IEnumerable<T> sequence)
+        {
+            // <SnippetTypePattern>
+            var third = sequence switch
+            {
+                System.Array array => array.GetValue(2),
+                IList<T> list => list[2],
+                IEnumerable<T> seq => seq.Skip(2).First(),
+            };
+            Console.WriteLine(third);
+            // </SnippetTypePattern>
+        }
+
+        private static void RecursiveExample<T>(IEnumerable<T> sequence)
+        {
+            // <SnippetRecursivePattern>
+            var third = sequence switch
+            {
+                System.Array { Length : 0}        => default,
+                System.Array { Length : 1} array => array.GetValue(0),
+                System.Array { Length : 2} array => array.GetValue(1),
+                System.Array array               => array.GetValue(2),
+                IList<T> list                    => list[2],
+                IEnumerable<T> seq               => seq.Skip(2).First(),
+            };
+            Console.WriteLine(third);
+            // </SnippetRecursivePattern>
+        }
+
+        private static void CaseGuardExample<T>(IEnumerable<T> sequence)
+        {
+            // <SnippetGuardCase>
+            var third = sequence switch
+            {
+                System.Array { Length : 0}                => default,
+                System.Array { Length : 1} array          => array.GetValue(0),
+                System.Array { Length : 2} array          => array.GetValue(1),
+                System.Array array                        => array.GetValue(2),
+                IEnumerable<T> list when !list.Any()      => default,
+                IEnumerable<T> list when list.Count() < 3 => list.Last(),
+                IList<T> list                             => list[2],
+                IEnumerable<T> seq                        => seq.Skip(2).First(),
+            };
+            Console.WriteLine(third);
+            // </SnippetGuardCase>
+        }
+
+        private static void ExhaustiveExample<T>(IEnumerable<T> sequence)
+        {
+            // <SnippetExhaustive>
+            var third = sequence switch
+            {
+                System.Array { Length : 0}       => default,
+                System.Array { Length : 1} array => array.GetValue(0),
+                System.Array { Length : 2} array => array.GetValue(1),
+                System.Array array               => array.GetValue(2),
+                IEnumerable<T> list 
+                    when !list.Any()             => default,
+                IEnumerable<T> list 
+                    when list.Count() < 3        => list.Last(),
+                IList<T> list                    => list[2],
+                null                             => throw new ArgumentNullException("null input"),
```

Comment:
In the case of [one-parameter constructor](https://docs.microsoft.com/en-us/dotnet/api/system.argumentnullexception.-ctor?view=netframework-4.8#System_ArgumentNullException__ctor_System_String_), the argument should be the name of a parameter:
```suggestion
                null                             => throw new ArgumentNullException(nameof(sequence)),
```